### PR TITLE
On Boarding UI

### LINKED ISSUE #13: Implement On-Boarding wizard view
- [x] Skip if user already went through the on-boarding process 
- [x] Implement navigation controls to go forward and backward the wizard
- [x] Implement the on-boarding screens
- [x] If application got closed/ crashed without completing the on-boarding screen, then it should start from the first page.
- [x] Even after updating to a new version, the application should skip this on-boarding screens on start.
- [x] Write unit test cases
- [x] Write e2e tests

> Note: Basic settings not a part of this task

### PR DESCRIPTION
**Pull request by shankar2105**
Resolve #13

### CODE REVIEW

File: app/store/addMiddlewares.ts [hunk 1]
```diff
@@ -20,4 +19,6 @@ export const addMiddlewares = ( middlewares: Array<Function> ): void => {
         // must be last
         middlewares.push( forwardToRenderer );
     }
+
+    middlewares.unshift( promiseMiddleware );
```
Comment (hunterlester) [inline app/store/addMiddlewares.ts]: Right, this is where I previously placed the `promiseMiddleware` in order to make promises properly resolve in redux actions, but then there seems to be a conflict with `electron-redux` [documentation](https://github.com/hardchor/electron-redux#install).
It looks like it only conflicts with `inRendererProcess` where `forwardToMain` must be the first middleware.
cc: @joshuef ^^

File: package.json [hunk 2]
```diff
@@ -34,8 +34,8 @@
     "start-renderer-dev": "cross-env NODE_ENV=development webpack-dev-server --config configs/webpack.config.renderer.dev.babel.js",
     "test": "cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 jest -u",
     "test-all": "yarn lint && yarn build && yarn test && yarn build-e2e && yarn test-e2e",
-    "test-e2e": "node -r babel-register-ts ./internals/scripts/CheckBuiltsExist.js && cross-env NODE_ENV=test TEST_CAFE=true testcafe electron:. ./__e2e__/*.e2e.ts",
-    "test-e2e-live": "node -r babel-register-ts ./internals/scripts/CheckBuiltsExist.js && cross-env NODE_ENV=test TEST_CAFE=true testcafe-live electron:. ./__e2e__/*.e2e.ts",
+    "test-e2e": "node -r babel-register-ts ./internals/scripts/CheckBuiltsExist.js && cross-env NODE_ENV=test TEST_CAFE=true testcafe electron:. ./__e2e__/*.e2e.ts --skip-js-errors",
```
Comment (hunterlester) [inline package.json]: Nice, thank you for including these for Linux.
They'll be obsolete once PR #38 is merged. <-- Let me know if you'd rather I rebased that PR on this `on_boarding` branch.

File: app/actions/helpers/preferences_db.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                this.tableName,
+                databaseCallBackHandler( resolver, reject )
+            );
+        } );
+    }
+
+    private updatePreferences( preferences: Preferences ) {
+        // eslint-disable-next-line consistent-return
+        return new Promise( async ( resolve, reject ) => {
+            if ( !this.preferenceId )
```
[the remark this answers is not in the document]
Comment (shankar2105) [reply app/actions/helpers/preferences_db.ts]: Let me check that

File: app/index.tsx [hunk 4]
```diff
@@ -27,3 +29,5 @@ if ( module.hot ) {
         );
     } );
 }
+
+store.dispatch( initialiseApp() );
```
Comment (joshuef) [inline app/index.tsx]: This should likely be done in another process

File: app/store/addMiddlewares.ts (hunk 1 above)
Comment (joshuef) [reply app/store/addMiddlewares.ts]: Hmmm, maybe `forwardToMain` will only work w/ resolved promises? 

A question: Should our render process be doing any `async` actions that may require a `promise` ?

**Comment by manavbp** [comment]
- [ ] Need to ensure that the store is being reset before each of the e2e tests

**Pull request closed without merging**